### PR TITLE
Support codeless non-dotnet functions

### LINKED ISSUE #6877: Support node worker for workflow extensions
cc @fabiocav  @hongzli

### PR DESCRIPTION
**Pull request by pragnagopa**
Resolves #6877

### CODE REVIEW

**Comment by pragnagopa** [comment]
@hongzli - verified changes in the PR with the repro you provided.

File: src/WebJobs.Script/Host/ScriptHost.cs [hunk 1]
```diff
@@ -304,7 +301,8 @@ public async Task InitializeAsync(CancellationToken cancellationToken = default)
 
                 // Initialize worker function invocation dispatcher only for valid functions after creating function descriptors
                 // Dispatcher not needed for non-proxy codeless function.
-                var filteredFunctionMetadata = functionMetadataList.Where(m => m.IsProxy() || !m.IsCodeless());
+                // Disptacher needed for non-dotnet codeless functions
+                var filteredFunctionMetadata = functionMetadataList.Where(m => m.IsProxy() || !Utility.IsCodelessDotNetLanguageFunction(m));
```
Comment (pragnagopa) [inline src/WebJobs.Script/Host/ScriptHost.cs:308]: @ankitkumarr - this allows loading "codeless" non-dotnet functions by the language worker. Let me know if anything else needs to be updated.

**Comment by pragnagopa** [comment]
gentle ping for review

**Comment by pragnagopa** [comment]
cc @hongzli - I believe you need this change

**Comment by pragnagopa** [comment]
@fabiocav / @ankitkumarr  - ping for review!

**Review by ankitkumarr — APPROVED** [review]

**Pull request merged**